fb/http.go

```diff
@@ -37,10 +43,41 @@ func NewFlashbladeClient(host string, insecure bool) *FlashbladeClient {
 
 	// Init x-auth-token
 	fb.refreshXAuthToken()
-
+	fb.ApiVersion = fb.getAPIVersion(version)
 	return &fb
 }
 
+func (fbClient *FlashbladeClient) getAPIVersion(selectedVersion string) string {
+	var (
+		params map[string]string
+		v      Version
+	)
+	err := fbClient.GetJSON("api_version", params, &v)
+	if err != nil {
+		log.Fatalln("Failed to get supported API-versions. Error =", err)
+	}
+
+	latestVersion := v.Versions[len(v.Versions)-1]
+
+	latest, err := strconv.ParseFloat(latestVersion, 32)
+	if err != nil {
+		log.Fatalf("Couldn't convert string to float. Err =", err)
+	}
+
+	selected, err := strconv.ParseFloat(selectedVersion, 32)
+	if err != nil {
+		log.Fatalf("Couldn't convert string to float. Err =", err)
+	}
+
+	apiVersion := strconv.FormatFloat(selected, 'f', -1, 32)
+
+	if latest < selected {
```

Comment:
A bit of an issue with treating the API version as a float is that "1.11 < 1.7", but 1.11 comes later in the release cycle! I think if we require 1.10 for anything then this behaviour is going to run into problems.

I'd suggest splitting on '.' and comparing the bits before and after the '.' as int.

Reply:
> 1.11 comes later in the release cycle

Oh I wasn't aware of this. I'll change it as you suggested